src/api/apiClients/superagent.ts

```diff
@@ -0,0 +1,69 @@
+import type { IApiClient, IRequestOptions, IResponse } from "../../types/api/apiClient.types.js";
+import superagent, { SuperAgentRequest, SuperAgentStatic } from "superagent";
+import { logApiStep } from "../../utils/reporter/decorators.js";
+
+class SuperagentApiClient implements IApiClient {
+  private agent: SuperAgentStatic;
+
+  constructor() {
+    this.agent = superagent.agent();
+  }
+  
+  @logApiStep
+  async sendRequest<T>(options: IRequestOptions): Promise<IResponse<T>> {
+    try {
+      const response = await this.createRequest(options).then((req) => req);
+      return this.transformResponse<T>(response);
+    } catch (error: any) {
+      console.log("Status:", error.response.status, "Request URL:", options.method, options.url);
+      console.log("Error:", error.response.text)
+      return this.transformResponse<T>(error.response);
+    }
+  }
+
+  @logApiStep
+  async sendFormDataRequest<T>(options: IRequestOptions): Promise<IResponse<T>> {
+    try {
+      const response = await this.createFormDataRequest(options);
+      return this.transformResponse<T>(response);
+    } catch (error: any) {
+      console.log("Status: ", error.response.status, "Request URL:", options.method, options.url);
+      console.log("Error: ", error.response.text)
+      return this.transformResponse<T>(error.response);
+    }
+  }
```

Comment:
that could be one method, it should follow DRY principle at least

Reply:
Removed this realization